### PR TITLE
Fix formatting of resource string where excess arguments are passed

### LINKED ISSUE #63607: Audit resource string formatting where excess arguments are passed
I ran the tests and dumped all the cases where a string was formatted with more arguments than format markers. Removed dupes and obvious cases where tests are doing the formatting and got the list below. List is in the format `|resource string| arg1 arg2 ...`. 

Many are bugs. Examples:

1. ArgumentException missing parameter name
`Buffer and BufferList properties cannot both be non-null.`. This corresponds to the resource string `net_ambiguousbuffers`. In the sources this is used like `throw new ArgumentException(SR.Format(SR.net_ambiguousbuffers, nameof(Buffer)));` which should be `throw new ArgumentException(SR.net_ambiguousbuffers, nameof(Buffer));`. Note there are three cases.

2. Bug in test
`Xslt_NoStylesheetLoaded` is formatted correctly in the product, but is passed too many arguments in the tests. This is low value to fix.

3. Message missing value it was intended to include
`The specified Oid is not valid. The Oid.FriendlyName or Oid.Value property must be set.` is used in code: `throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_InvalidCurveOid, curve.Oid.Value));`  The fix is possibly to change the string to add a format marker so that the value appears, and then fix up code that doesn't pass one, eg `throw new ArgumentException(SR.Cryptography_InvalidCurveOid);`

The task here is to run through the list and fix as appropriate. 

```
|'{0}' from assembly '{1}' has too many dimensions.| `System.Int32[,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,]`  `System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`  ``
|'{1}' is an invalid value for the '{0}' attribute.| `form`  `qualifieds`  ``
|A null or disposed certificate was present in CustomTrustStore.| `CustomTrustStore`
|All field widths, except the last element, must be greater than zero. A field width less than or equal to zero in the last element indicates the last field is of variable length.| `FieldWidths`
|An undefined NumberStyles value is being used.| `style`
|Buffer and BufferList properties cannot both be non-null.| `Buffer`
|Buffer and BufferList properties cannot both be non-null.| `BufferList`
|Collectible type '{0}' has unsupported FixedAddressValueTypeAttribute applied to a field.| `System.Runtime.Loader.Tests.TestClassNotSupported_FixedAddressValueType`  `System.Runtime.Loader.Test.AssemblyNotSupported, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`  ``
|Could not create array type '{0}' from assembly '{1}' because the element type is ByRef-like.| `System.TypedReference[]`  `System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`  ``
|Could not create array type '{0}' from assembly '{1}' because the element type is ByRef.| `System.Int32&`  `System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`  ``
|Could not load interface '{0}' from assembly '{1}' because it must extend from Object.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Could not load type '{0}' from assembly '{1}' because it attempts to implement a class as an interface.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Could not load type '{0}' from assembly '{1}' because the format is invalid.| `Invalid_Token.0x1B000000`  `System.Runtime.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`  ``
|Could not load type '{0}' from assembly '{1}' because the parent type is sealed.| `NestedType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Could not load type '{0}' from assembly '{1}'.| `A.B`  `System.Reflection.Emit.Lightweight.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`  ``
|Could not load TypeSpec from assembly '{1}'.| `Invalid_Token.???[`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Current size cannot be determined for this store.| `CurrentSize`
|Custom trust certificates were provided while in System trust mode.| `TrustMode`
|Event IDs must be positive integers.| `WriteInteger`
|Event {0} has ID {1} which is already in use.| `SetActivityId`  `40`  `TransactionstateEnlist`
|Expression must evaluate to a node-set.| `/bookstore/child::(book | magazine)`
|Invalid document name.| `128`
|Invalid Field Access Flags.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Invalid file format.| `/idontexist.xml`
|IPv4 address 0.0.0.0 and IPv6 address ::0 are unspecified addresses that cannot be used as a target address.| `address`
|Load factor needs to be between 0.1 and 1.0.| `0.1`  `1`
|More than one export was found that matches the constraint:
|Native type '{0}' has a constructor taking a caller-allocated buffer, but does not support marshalling in scenarios where using a caller-allocated buffer is impossible| `Native`  `S`
|No exports were found that match the constraint:
|No stylesheet was loaded.| `IDontExist.xsl`
|Non-negative number required.| `maxValue`
|NumberOfChars must be greater than zero.| `numberOfChars`
|Out of range.| `IntRange`  `-5`  `5`
|Property, indexer, or event '{0}' is not supported by the language; try directly calling accessor method '{1}'| `ItemProp`  `TestType.set_ItemProp(int, int)`  ``
|Property, indexer, or event '{0}' is not supported by the language; try directly calling accessor methods '{1}' or '{2}'| `ItemProp`  `TestType.get_ItemProp(int)`  `TestType.set_ItemProp(int, int)`  ``
|Registry key name must start with a valid base key name.| `keyName`
|Signature of the body and declaration in a method implementation do not match.  Type: '{0}'.  Assembly: '{1}'.| `_proxy_System.ComponentModel.Composition.ITrans_MetadataTests_MetadataViewWithIndexer_b587effc-bc4a-464b-a7f7-088d17d9279e`  `MetadataViewProxies_7d8cd428-d5fb-41b8-a2de-3df47e34980f, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Stream was not readable.| `stream`
|Stream was not writable.| `stream`
|String contains invalid Unicode code points.| `replacement`
|Text "{0}" cannot be parsed. The expected text format is "{1}".| `text`  `1, 2, 3`  `x, y, width, height`
|The Buffer space specified by the Count property is insufficient for the AcceptAsync method.| `Count`
|The field {0} must be a string with a maximum length of {1}.| `Object`  `12`  `0`
|The given key '{0}' was not present in the dictionary.| `0`  `abc`
|The maximum read depth ({0}) has been exceeded because XML data being read has more levels of nesting than is allowed by the quota. This quota may be increased by changing the MaxDepth property on the XmlDictionaryReaderQuotas object used when creating the XML reader.| `1`  ``  ``
|The maximum string content length quota ({0}) has been exceeded while reading XML data. This quota may be increased by changing the MaxStringContentLength property on the XmlDictionaryReaderQuotas object used when creating the XML reader.| `1024`  ``  ``
|The non-generic {1} '{0}' cannot be used with type arguments| `Dynamic.Tests.ExplicitlyImplementedInterface2.Foo()`  `method`  `0`
|The requested key container was not found.| `Error`
|The signature is incorrect.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|The specified Oid is not valid. The Oid.FriendlyName or Oid.Value property must be set.| `1.2.840.10045.3.0.1`
|The type of {0} is not expected in {1}.| `Op1`  `EventKeywords`  `Opcodes`
|The value must be different than {0}.| `1`  `1`
|The value must be equal to {0}.| `1`  `1`
|The Whitespace or SignificantWhitespace node can contain only XML whitespace characters. '{0}' is not an XML white space character.| `.`  `0x00`
|There are not enough bytes remaining in the accessor to read at this position.| `System.IO.Tests.Uma_TestStructs+UmaTestStruct`
|There are not enough bytes remaining in the accessor to write at this position.| `System.IO.Tests.Uma_TestStructs+UmaTestStruct`
|Type '{0}' from assembly '{1}' has a field of an illegal type.| `Name`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Type '{0}' has a 'GetPinnableReference' method but its native type does not support marshalling in scenarios where pinning is impossible| `S`  `Native`
|Unable to read delimited fields because Delimiters is Nothing or empty.| `Delimiters`
|Unexpected XML declaration. The XML declaration must be the first node in the document, and no whitespace characters are allowed to appear before it.| `xml`
|XSLT compile error.| ``  `0`  `0`
```

### PR DESCRIPTION
**Pull request by Marusyk**
Only the lines with `+` were fixed
```
|'{0}' from assembly '{1}' has too many dimensions.| `System.Int32[,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,]`  `System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`  ``
|'{1}' is an invalid value for the '{0}' attribute.| `form`  `qualifieds`  ``
+	|A null or disposed certificate was present in CustomTrustStore.| `CustomTrustStore`
+	|All field widths, except the last element, must be greater than zero. A field width less than or equal to zero in the last element indicates the last field is of variable length.| `FieldWidths`
+	|An undefined NumberStyles value is being used.| `style`
+	|Buffer and BufferList properties cannot both be non-null.| `Buffer`
+	|Buffer and BufferList properties cannot both be non-null.| `BufferList`
|Collectible type '{0}' has unsupported FixedAddressValueTypeAttribute applied to a field.| `System.Runtime.Loader.Tests.TestClassNotSupported_FixedAddressValueType`  `System.Runtime.Loader.Test.AssemblyNotSupported, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`  ``
|Could not create array type '{0}' from assembly '{1}' because the element type is ByRef-like.| `System.TypedReference[]`  `System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`  ``
|Could not create array type '{0}' from assembly '{1}' because the element type is ByRef.| `System.Int32&`  `System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`  ``
|Could not load interface '{0}' from assembly '{1}' because it must extend from Object.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Could not load type '{0}' from assembly '{1}' because it attempts to implement a class as an interface.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Could not load type '{0}' from assembly '{1}' because the format is invalid.| `Invalid_Token.0x1B000000`  `System.Runtime.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`  ``
|Could not load type '{0}' from assembly '{1}' because the parent type is sealed.| `NestedType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Could not load type '{0}' from assembly '{1}'.| `A.B`  `System.Reflection.Emit.Lightweight.Tests, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51`  ``
|Could not load TypeSpec from assembly '{1}'.| `Invalid_Token.�?�??[`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
+	|Current size cannot be determined for this store.| `CurrentSize`
+	|Custom trust certificates were provided while in System trust mode.| `TrustMode`
+	|Event IDs must be positive integers.| `WriteInteger`
+	|Event {0} has ID {1} which is already in use.| `SetActivityId`  `40`  `TransactionstateEnlist`
|Expression must evaluate to a node-set.| `/bookstore/child::(book | magazine)`
+	|Invalid document name.| `128`
|Invalid Field Access Flags.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
+	|Invalid file format.| `/idontexist.xml`
+	|IPv4 address 0.0.0.0 and IPv6 address ::0 are unspecified addresses that cannot be used as a target address.| `address`
+	|Load factor needs to be between 0.1 and 1.0.| `0.1`  `1`
|More than one export was found that matches the constraint:
|Native type '{0}' has a constructor taking a caller-allocated buffer, but does not support marshalling in scenarios where using a caller-allocated buffer is impossible| `Native`  `S`
|No exports were found that match the constraint:
+	|No stylesheet was loaded.| `IDontExist.xsl`
+	|Non-negative number required.| `maxValue`
+	|NumberOfChars must be greater than zero.| `numberOfChars`
|Out of range.| `IntRange`  `-5`  `5`
|Property, indexer, or event '{0}' is not supported by the language; try directly calling accessor method '{1}'| `ItemProp`  `TestType.set_ItemProp(int, int)`  ``
|Property, indexer, or event '{0}' is not supported by the language; try directly calling accessor methods '{1}' or '{2}'| `ItemProp`  `TestType.get_ItemProp(int)`  `TestType.set_ItemProp(int, int)`  ``
+	|Registry key name must start with a valid base key name.| `keyName`
|Signature of the body and declaration in a method implementation do not match.  Type: '{0}'.  Assembly: '{1}'.| `_proxy_System.ComponentModel.Composition.ITrans_MetadataTests_MetadataViewWithIndexer_b587effc-bc4a-464b-a7f7-088d17d9279e`  `MetadataViewProxies_7d8cd428-d5fb-41b8-a2de-3df47e34980f, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Stream was not readable.| `stream`
|Stream was not writable.| `stream`
+	|String contains invalid Unicode code points.| `replacement`
+	|Text "{0}" cannot be parsed. The expected text format is "{1}".| `text`  `1, 2, 3`  `x, y, width, height`
+	|The Buffer space specified by the Count property is insufficient for the AcceptAsync method.| `Count`
|The field {0} must be a string with a maximum length of {1}.| `Object`  `12`  `0`
|The given key '{0}' was not present in the dictionary.| `0`  `abc`
|The maximum read depth ({0}) has been exceeded because XML data being read has more levels of nesting than is allowed by the quota. This quota may be increased by changing the MaxDepth property on the XmlDictionaryReaderQuotas object used when creating the XML reader.| `1`  ``  ``
|The maximum string content length quota ({0}) has been exceeded while reading XML data. This quota may be increased by changing the MaxStringContentLength property on the XmlDictionaryReaderQuotas object used when creating the XML reader.| `1024`  ``  ``
|The non-generic {1} '{0}' cannot be used with type arguments| `Dynamic.Tests.ExplicitlyImplementedInterface2.Foo()`  `method`  `0`
+	|The requested key container was not found.| `Error`
|The signature is incorrect.| `TestType`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
+	|The specified Oid is not valid. The Oid.FriendlyName or Oid.Value property must be set.| `1.2.840.10045.3.0.1`
+	|The type of {0} is not expected in {1}.| `Op1`  `EventKeywords`  `Opcodes`
|The value must be different than {0}.| `1`  `1`
||The value must be equal to {0}.| `1`  `1`
|The Whitespace or SignificantWhitespace node can contain only XML whitespace characters. '{0}' is not an XML white space character.| `.`  `0x00`
+	|There are not enough bytes remaining in the accessor to read at this position.| `System.IO.Tests.Uma_TestStructs+UmaTestStruct`
+	|There are not enough bytes remaining in the accessor to write at this position.| `System.IO.Tests.Uma_TestStructs+UmaTestStruct`
|Type '{0}' from assembly '{1}' has a field of an illegal type.| `Name`  `TestAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null`  ``
|Type '{0}' has a 'GetPinnableReference' method but its native type does not support marshalling in scenarios where pinning is impossible| `S`  `Native`
+	|Unable to read delimited fields because Delimiters is Nothing or empty.| `Delimiters`
|Unexpected XML declaration. The XML declaration must be the first node in the document, and no whitespace characters are allowed to appear before it.| `xml`
|XSLT compile error.| ``  `0`  `0`
```
Fixes #63607

Please review
Thank you in advance

### CODE REVIEW

**Comment by danmoseley** [comment]
Let me know when it's passing and ready for review 🙂

File: src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/Registry.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ private static RegistryKey GetBaseKeyFromKeyName(string keyName, out string subK
                 return baseKey;
             }
 
-            throw new ArgumentException(SR.Format(SR.Arg_RegInvalidKeyName, nameof(keyName)), nameof(keyName));
+            throw new ArgumentException(SR.Arg_RegInvalidKeyName, nameof(keyName));
```
Comment (danmoseley) [inline src/libraries/Microsoft.Win32.Registry/src/Microsoft/Win32/Registry.cs:67]: We can change resource strings if we want. In some cases, perhaps including more info might be worthwhile. Eg.
```c#
throw new ArgumentException(SR.Format(SR.Arg_RegInvalidKeyName, keyName), nameof(keyName));
```
with maybe
```
Registry key name "{0}" must start with a valid base key name.
```
up to you.

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/FileFormatException.cs [hunk 2]
```diff
@@ -55,7 +55,7 @@ public FileFormatException(Uri? sourceUri)
             : base(
                 sourceUri == null
                 ? SR.FileFormatException
-                : SR.Format(SR.FileFormatExceptionWithFileName, sourceUri))
+                : SR.FileFormatExceptionWithFileName)
```
Comment (danmoseley) [inline src/libraries/System.IO.Packaging/src/System/IO/Packaging/FileFormatException.cs:58]: this is a case where the resource is clearly wrong. both these resources are the same string.

probably the right fix for these two is to change the resource to eg `File "{0}" has an invalid file format.` or similar, and leave the sourceUri in it.

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs [hunk 3]
```diff
@@ -3501,7 +3501,7 @@ private static void AddProviderEnumKind(ManifestBuilder manifest, FieldInfo stat
 #endif
             return;
             Error:
-            manifest.ManifestError(SR.Format(SR.EventSource_EnumKindMismatch, staticField.Name, staticField.FieldType.Name, providerEnumKind));
+            manifest.ManifestError(SR.Format(SR.EventSource_EnumKindMismatch, staticField.FieldType.Name, providerEnumKind));
```
Comment (danmoseley) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:3504]: @dotnet/area-system-diagnostics-tracesource could you please review the changes in /Tracing/ ? It's not clear to me which 2 of the 3 parameters are supposed to go into the string. Today it's implicitly the first 2.

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECCurve.cs [hunk 4]
```diff
@@ -194,10 +194,14 @@ public void Validate()
                     throw new CryptographicException(SR.Cryptography_InvalidECNamedCurve);
                 }
 
-                if (Oid == null ||
-                    (string.IsNullOrEmpty(Oid.FriendlyName) && string.IsNullOrEmpty(Oid.Value)))
+                if (Oid == null)
```
Comment (danmoseley) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECCurve.cs:197]: I'd rather continue to throw CryptographicException here. changing to ArgumentNullException would be a breaking change (which we sometimes make) but changing to ArgumentException would definitely be unacceptable.

**Comment by danmoseley** [comment]
Thanks @Marusyk changes look good with a few comments.

**Comment by danmoseley** [comment]
(BTW, if you make changes please add as new commits rather than rebase/ force push so I don't have to re-review everything.)

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs (hunk 3 above)
Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:3504]: @brianrob can you page in enough context briefly to confirm that the two changes in this file are correct? The 2nd one just preserves existing behavior, although that might be wrong.

After that this is good to merge.

Comment (brianrob) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:3504]: Yup, the changes in `EventSource.cs` look good to me.  It looks like there were additional parameters passed to `SR.Format` that should never have been there.

Comment (danmoseley) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:3504]: thanks @brianrob !

**Review by danmoseley — APPROVED** [review]

**Comment by danmoseley** [comment]
Thanks @Marusyk .. this closes the issue, so I assume that means you don't think there were others there worth fixing (which is fine)

**Pull request merged**